### ISSUE TITLE
About pose variation in VAE face reconstruction examples

### ISSUE THREAD

**Issue by draveos** [PiLab-CAU/ImageProcessing-2502#18]
I have a question about the VAE Face Reconstruction and VAE Face Generation examples in Lecture 9. In the slides, most of the faces look like they are close to a frontal pose. (e.g., the "VAE Face Reconstruction" and "VAE Face Generation" figures)

<img width="871" height="489" alt="Image" src="https://github.com/user-attachments/assets/6036edfc-7e0e-41f1-84d2-199ab7d06850" />
<img width="813" height="469" alt="Image" src="https://github.com/user-attachments/assets/6aac3da9-c545-4bc2-9fb6-0d441d1be23a" />

From what I understand, these models are usually trained on face datasets such as CelebA, where most images are roughly frontal, and extreme poses (strong profile views, looking up/down, etc.) are relatively rare.

My questions are:

1. In the examples shown in the slides, does the training dataset actually contain a wide range of head poses (left/right profile, up/down), and the VAE successfully learns to reconstruct them as well?  Or are the examples mainly focusing on near-frontal faces because that is what the dataset mostly contains?

2. If the dataset does include more extreme poses but the VAE reconstructions tend to “pull” them toward a more frontal and averaged-looking face, should we interpret this as:
   - a sign of underfitting / insufficient training, or
   - a more fundamental behavior of VAEs (due to the Gaussian prior and KL regularization) that tends to smooth out rare variations in the data?

Thank you.

**Comment by yjyoo3312** [comment]
@draveos 

You’re reading the situation correctly, and it’s good that you’re separating “what’s in the dataset” from “what the VAE objective prefers.”

First, I'd like to ask you about the examples on the slides. In almost all teaching demos (and in many papers) the face VAEs are trained on CelebA or something very similar. Those datasets are heavily biased toward roughly frontal, well-aligned faces. There are some slight yaw and pitch variations, but true extreme poses (complete profile, strong up-/down-tilt, occlusions) are relatively rare. The fact that most reconstructions and generations shown in the slides look near-frontal mainly reflects the dataset distribution rather than demonstrating that the model robustly handles every possible head pose. If you train a standard convolutional VAE on CelebA and just randomly sample reconstructions and generations, your slide will naturally fill with near-frontal faces, because that is what the dataset is dominated by.

Now, suppose you do have a dataset with more extreme poses, or you look carefully at the few extreme-pose samples in CelebA. What typically happens with a “vanilla” VAE (Gaussian prior, KL term with β≈1) is that the reconstructions of those rare poses are not catastrophically wrong. Still, they are noticeably “pulled” back toward the mean manifold: the jaw may rotate a bit toward frontal, eyes become more symmetric, background and hair become more generic, and the overall face looks more average than the input. This is not just sloppy training; it is very much tied to how the VAE objective works.

The VAE balances two pressures: pixel-level reconstruction accuracy and KL regularization, which aims to keep the approximate posterior q(z|x) close to the simple prior p(z) (usually N(0,I)). For common, high-density regions of the data (frontal faces), the model can afford to carve out a reasonably rich region of latent space that captures variations in identity, expression, lighting, etc., while keeping q(z|x) not too far from N(0,I). For rare configurations like extreme poses, the optimizer has a more complex trade-off: dedicating a lot of latent volume and decoder capacity to reconstruct those samples perfectly increases reconstruction accuracy on a tiny fraction of the dataset, but it also makes q(z|x) more irregular and more expensive under the KL term. Because the objective is an average over the whole dataset, it is “cheaper” to partially sacrifice those rare images by mapping them to latent codes closer to the mass of the prior and letting the decoder output something more typical. That is precisely the mode-covering, smoothing behavior you see in VAEs: they prefer to blur or average unusual examples rather than create isolated sharp modes for them.

So when you observe a rare pose being reconstructed as a more frontal, averaged face, the correct interpretation is not simply “the model is underfit, train longer” (although underfitting can undoubtedly make it worse). It is essentially a consequence of the VAE’s Gaussian prior and KL regularization: the model is encouraged to pull posteriors toward a simple, unimodal latent distribution, thereby compressing rare variations into nearby typical codes. If you reduce the KL weight (β<1), increase the latent dimensionality, or switch to a more flexible prior (mixture, VampPrior, hierarchical VAEs), those rare poses are often reconstructed more faithfully, but at the cost of a less “nice” latent space or weaker generative behavior. Conversely, if you push β>1 or enforce stronger regularization for nicer latent structure, the averaging effect becomes even more pronounced.

In that sense, the lecture slides use “face VAEs on CelebA” in its friendliest regime: the data already live near a low-dimensional, aligned, frontal manifold, and the standard Gaussian-prior VAE can model that reasonably well. The slides do not claim that the same simple VAE would perfectly reconstruct all extreme poses; instead, they illustrate that, given a biased dataset, the VAE learns a low-dimensional latent space whose typical samples resemble the typical images in the dataset. Your way of thinking about it is precisely correct: the dominance of near-frontal faces comes from the dataset, and the tendency to pull unusual examples toward something more frontal and average is a structural behavior of VAEs under Gaussian priors and KL regularization, which can be mitigated but not completely avoided without changing the objective or the prior.
